pumpkin-solver/src/statistics/statistic_logging.rs

```diff
@@ -41,14 +43,16 @@ pub fn configure_statistic_logging(
 
 /// Logs the provided statistic with name `name` and value `value`. At the moment it will log in
 /// the format `STATISTIC_PREFIX NAME=VALUE`.
-pub fn log_statistic(name: impl Display, value: impl Display) {
+pub fn write_statistic(writer: &mut Box<dyn Write>, name: impl Display, value: impl Display) {
     if let Some(statistic_options) = STATISTIC_OPTIONS.get() {
         let name = if let Some(casing) = &statistic_options.statistics_casing {
             name.to_string().to_case(*casing)
         } else {
             name.to_string()
         };
-        println!("{} {name}={value}", statistic_options.statistic_prefix)
+        if let Err(e) = writer.write_fmt(format_args!("{} {name}={value}\n", statistic_options.statistic_prefix)) {
```

Comment:
I think we can use the `write!` macro directly here

Reply:
Indeed! It compiles to exactly the same